### ISSUE TITLE
Why No Softmax?

### ISSUE THREAD

**Issue by kingnobro** [lucidrains/DALLE-pytorch#444]
In `class DALLE(nn.Module)`, there is a member called `to_logits`,

```python
self.to_logits = nn.Sequential(
    nn.LayerNorm(dim),
    nn.Linear(dim, self.total_tokens),
)
```

Why there is no `Softmax` after `nn.Linear`? I read the paper _Attention Is All You Need_, and there is a softmax function after linear layer.
If there is **no** softmax, the value in logits might be very big. So in function `generate_images`, when passing the logits containing a very big number to function `gumbel_sample`, the uniform noise cannot influence the sample result.

**Comment by alexriedel1** [comment]
https://github.com/lucidrains/DALLE-pytorch/blob/daf30d02a42d4e407b540a568950cdf085f5d9b1/dalle_pytorch/dalle_pytorch.py#L652

`torch.nn.CrossEntropyLoss``is a combination of softmax and NLL Loss

`https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss`
